common/rpc/grpc.go

```diff
@@ -181,20 +185,51 @@ func NewServiceErrorInterceptor(
 		}
 		return resp, serviceerror.ToStatus(err).Err()
 	}
+}
+
+func NewFrontendErrorInterceptor(
+	logger log.Logger, hideErrorsAtFrontend bool,
+) grpc.UnaryServerInterceptor {
+	return func(
+		ctx context.Context,
+		req interface{},
+		_ *grpc.UnaryServerInfo,
+		handler grpc.UnaryHandler,
+	) (interface{}, error) {
 
+		resp, err := handler(ctx, req)
+		if err != nil && hideErrorsAtFrontend {
+			err = HideUnknownOrInternalErrors(logger, err)
+		}
+		return resp, serviceerror.ToStatus(err).Err()
+	}
 }
 
-func FrontendErrorInterceptor(
-	ctx context.Context,
-	req interface{},
-	_ *grpc.UnaryServerInfo,
-	handler grpc.UnaryHandler,
-) (interface{}, error) {
-	resp, err := handler(ctx, req)
+var errorFrontendUnknown = errors.New("something went wrong, please retry")
 
-	// mask some internal errors at frontend
+func HideUnknownOrInternalErrors(
+	logger log.Logger,
+	err error,
+) error {
+	// convert some internal errors into specific errors
 	if _, ok := err.(*serviceerrors.ShardOwnershipLost); ok {
-		err = serviceerror.NewUnavailable("shard unavailable, please backoff and retry")
+		return serviceerror.NewUnavailable("shard unavailable, please backoff and retry")
```

Comment:
Why did you leave `ShardOwnershipLost` here?

Reply:
removed in the next iter.

Reply:
Can you please verify that shard ownership lost error is masked by the new logic in this PR (A unit test would be good). Earlier, it was specially handled to mask that error message.

Reply:
ok, add it back and add tests for it

Reply:
This interceptor wasn't used before. It might be masked somewhere else, but not here. I don't like some exceptional case to be here. All `serviceerrors` (with `s`, meaning "server") should not leave FE. So you either mask all of them, or don't mask anything because we already did it at other places.

Reply:
discussed offline. This is handled by a separate interceptors